### PR TITLE
Closes #343: Persist FxA state in shared prefs

### LINKED ISSUE #343: Persist and restore fxa state to avoid having to log in every time
Currently a log in is required every time the app starts.

### PR DESCRIPTION
**Pull request by carolkng**
r? @csadilek

### CODE REVIEW

File: samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt [hunk 1]
```diff
@@ -27,19 +28,40 @@ open class MainActivity : AppCompatActivity() {
...
         }
 
         findViewById<View>(R.id.button).setOnClickListener {
             account?.beginOAuthFlow(scopes, false)?.whenComplete { openTab(it) }
         }
+
+        findViewById<View>(R.id.buttonPurge).setOnClickListener {
+            getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE).edit().putString(FXA_STATE_KEY, "").apply()
+            val txtView: TextView = findViewById(R.id.txtView)
+            txtView.text = "State cleared!"
```
Comment (csadilek) [inline samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt]: We need to put "State cleared!" into strings.xml

File: samples/firefox-accounts/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -6,4 +6,5 @@
     <string name="app_name">FxA Android Demo</string>
     <string name="sign_in">FxA Sign In</string>
     <string name="signed_in">Signed in: %1$s</string>
+    <string name="purge">Purge state</string>
```
Comment (csadilek) [inline samples/firefox-accounts/src/main/res/values/strings.xml]: Should we maybe just call this "sign out" or "logout"?

Comment (carolkng) [reply samples/firefox-accounts/src/main/res/values/strings.xml]: updated to "FxA Log Out"

File: samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt (hunk 1 above)
Comment (carolkng) [reply samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt]: Updated to "Logged out!" to be consistent with the button

**Pull request merged**